conduit/data/datasets/audio/ecoacoustics.py

```diff
@@ -299,10 +295,11 @@ def _preprocess_audio(self) -> None:
                     f"(fractional remainder must be greater than 0.5): discarding terminal segment."
                 )
                 waveform = waveform[
-                    :, : num_segments * self.specgram_segment_len * self.resample_rate
+                    :, : int(num_segments * self.specgram_segment_len * self.resample_rate)
                 ]
 
             waveform_segments = waveform.chunk(chunks=num_segments, dim=-1)
+            assert self.transform is not None
```

Comment:
transform shouldn't be optional, or at least, if it is, the preprocessing shouldn't be run and the original wav files should be used.